app/modules/scan/service.server.ts

```diff
@@ -138,3 +145,61 @@ export async function getScanByQrId({ qrId }: { qrId: string }) {
     });
   }
 }
+
+export async function createScanNote({ userId, qrId, latitude, longitude, manuallyGenerated }: {
+  userId?: string | null;
+  qrId: string;
+  latitude?: Scan["latitude"];
+  longitude?: Scan["longitude"];
+  manuallyGenerated?: boolean;
+}) {
+  try {
+    let message = "";
+    const { assetId, organizationId } = await getQr(qrId)
+    if (assetId) {
+      if (userId && userId != "anonymous") {
+        const user = await db.user.findFirst({
```

Comment:
Okey a few things.
1. IN general we dont do those inline queries anymore, because we need to try/catch them every time so its a lot of repeating code.
2. You should simply use `getUserByID` as that already handles catching errors and so on.